### PR TITLE
Improve travis targets

### PR DESCRIPTION
**Pull request by ruflin**
* Explicitly define all targets
* Use environment variable for Golang version
* Update docker-compose version to 1.9.0
* Add os x targets for heartbeat
* Remove docker update as not needed anymore
* Add heartbeat coverage report

### CODE REVIEW

**Comment by ruflin** [comment]
@andrewkroh The jenkins builds fail now, as they rely on the entry in the travis file. I suggest we move the golang version the version.asciidoc file and read it from there.

**Comment by ruflin** [comment]
jenkins, retest it

File: .travis.yml [hunk 1]
```diff
@@ -5,71 +5,80 @@ services:
...
+    - &go_version 1.7.1
 
 matrix:
-  exclude:
-    - os: osx
+  include:
+    # General checks
+    - os: linux
       env: TARGETS="check"
+      go: *go_version
```
Comment (andrewkroh) [inline .travis.yml]: If `go_version` is global how come it must be set everywhere in the matrix?

Comment (ruflin) [inline .travis.yml]: Unfortunately this mix of global go version and include doesn't work :-( It will not read the global golang version.

**Pull request merged**